cogs/mod.py

```diff
@@ -309,10 +310,24 @@ def check_if_target_is_staff(self, target):
         log_channel = self.bot.get_channel(config.modlog_channel)
         if not channel:
             channel = ctx.channel
-        await channel.purge(limit=limit)
-        msg = f"🗑 **Purged**: {ctx.author.mention} purged {limit} "\
+        purge = await channel.purge(limit=limit)
+        newLimit = len(purge)
+        msg = f"🗑 **Purged**: {ctx.author.mention} purged {newLimit} "\ # More accurate to check for the length of the purged list.
               f"messages in {channel.mention}."
+        # Open purge_list.txt (Or create if doesn't exist)
+        f = open("purge_list.txt","w+")
+        # Loop through all messages in the purged messages
+        for m in purge:
+            mc = str(m.clean_content).encode("utf8")
+            f.write("MSG ID: " + str(m.id) + " | Author Name and Hash: " + str(m.author.name) + "#" + str(m.author.discriminator) + " |  Timestamp: " +str(m.created_at)+" | Message Content: " + str(mc).split("b'",1)[1] +" |\n")
```

Comment:
Can we have these in a single line, like an IRC output?

Also, what the heck are you doing with the message content, just use message's clean_content, it should be a string (which should play nicely with stringio).

Also, just put the whole thing in an fstring.

Also, just use `str(m.author)` instead of `str(m.author.name) + "#" + str(m.author.discriminator)`.

Reply:
Yea, I encoded the string to utf8 cause the emoji's are fucking it up.

Reply:
What do you mean IRC output, If we put it all in one line. like without looping through contents.
like if you just do print(purge) ?

Reply:
Would it better not do it?, for comfort and easier reading?

Reply:
No, like `[$snowflake] <$nameanddiscriminator ($id)>: $message`

Reply:
wdym, this [$snowflake] <$nameanddiscriminator ($id)>: $message repeating like in the same line? or like this
[$snowflake] <$nameanddiscriminator ($id)>: $message
[$snowflake] <$nameanddiscriminator ($id)>: $message

Reply:
Instead of snowflake wouldn't it be better to use the time stamp?

Reply:
one line per message. should be simple enough.

Reply:
snowflake makes it easier to file reports if needed.

Reply:
So the second option, the one where its like this
[$snowflake] <$nameanddiscriminator ($id)>: $message
[$snowflake] <$nameanddiscriminator ($id)>: $message

Reply:
Snowflake is the ID of message, figured it out rn